src/PolarCircleResolution.js

```diff
@@ -0,0 +1,119 @@
+  }
+  return false;
+}
+
+const aqrabYaumResolver = (coordinates, date, daysAdded) => {
+  if (daysAdded <= Math.ceil(365 / 2)) {
+    const pastDate = new Date(date.getTime());
+    pastDate.setDate(pastDate.getDate() - daysAdded);
+    const solarTimePast = new SolarTime(pastDate, coordinates);
+    const solarTimePastTomorrow = new SolarTime(dateByAddingDays(pastDate, 1), coordinates);
+
+    if (!isValidSolarTime(solarTimePast) || !isValidSolarTime(solarTimePastTomorrow)) {
+      const futureDate = new Date(date.getTime());
+      futureDate.setDate(futureDate.getDate() + daysAdded);
+      const solarTimeFuture = new SolarTime(futureDate, coordinates);
+      const solarTimeFutureTomorrow = new SolarTime(dateByAddingDays(futureDate, 1), coordinates);
+
+      if (!isValidSolarTime(solarTimeFuture) || !isValidSolarTime(solarTimeFutureTomorrow)) {
+        return aqrabYaumResolver(coordinates, date, daysAdded + 1);
+      } else {
+        return {
+          solarTime: solarTimeFuture,
+          tomorrowSolarTime: solarTimeFutureTomorrow,
+          date: futureDate,
+        }
+      }
+    } else {
+      return {
+        solarTime: solarTimePast,
+        tomorrowSolarTime: solarTimePastTomorrow,
+        date: pastDate,
+      }
+    }
+  }  
+  return false;
+}
+
+const aqrabBaladResolver = (coordinates, date, latitude) => {
+  const solarTime = new SolarTime(date, { ...coordinates, latitude });
+  const tomorrowSolarTime = new SolarTime(dateByAddingDays(date, 1), { ...coordinates, latitude });
+  if (!isValidSolarTime(solarTime) || !isValidSolarTime(tomorrowSolarTime)) {
+    if (Math.abs(latitude) >= UNSAFE_LATITUDE) {
+      return aqrabBaladResolver(coordinates, date, latitude - Math.sign(latitude) * LATITUDE_VARIATION_STEP);
+    }
+    return false;
+  }  
+  return {
+    solarTime,
+    tomorrowSolarTime,
+    latitude,
+  };
+};
+
+export const polarCircleResolvedValues = (resolver, date, coordinates) => {
+  let resolvedLatitude = coordinates.latitude;
+  let resolvedDate = new Date(date.getTime());
+  let resolvedSolarTime = null;
+  let resolvedTomorrowSolarTime = null;
+  
+  const defaultReturn = {
+    date,
+    coordinates,
+    resolvedSolarTime: new SolarTime(date, coordinates),
+    resolvedTomorrowSolarTime: new SolarTime(dateByAddingDays(date, 1), coordinates),
+  };
+
+  switch (resolver) {
+    case PolarCircleResolution.aqrabYaum: {
+      const resolved = aqrabYaumResolver(coordinates, date, 1);
+      if (resolved) {
+        resolvedDate = resolved.date;
+        resolvedSolarTime = resolved.solarTime;
+        resolvedTomorrowSolarTime = resolved.tomorrowSolarTime;
+      } else {
+        return defaultReturn;
+      }
+      break;
+    }
+    case PolarCircleResolution.aqrabBalad: {
+      const resolved = aqrabBaladResolver(coordinates, date, resolvedLatitude - (Math.sign(resolvedLatitude) * LATITUDE_VARIATION_STEP));
+      if (resolved) {
```

Comment:
If we have the individual resolver functions have the same return values as this function then we can simplify this to be

`return resolved | defaultReturn`